api/client/src/main/java/org/projectnessie/client/http/HttpClientBuilder.java

```diff
@@ -323,12 +337,18 @@ public <API extends NessieApi> API build(Class<API> apiVersion) {
     if (apiVersion.isAssignableFrom(HttpApiV1.class)) {
       builder.setJsonView(Views.V1.class);
       HttpClient httpClient = builder.build();
+      if (enableApiCompatibilityCheck) {
+        NessieApiCompatibility.check(1, httpClient);
+      }
       return (API) new HttpApiV1(new NessieHttpClient(httpClient));
     }
 
     if (apiVersion.isAssignableFrom(HttpApiV2.class)) {
```

Comment:
Can you move this `if` up (minor nit)

Reply:
Actually this would break things:  `HttpApiV2` is a subclass of `NessieApiV1`.